### PR TITLE
.Net & Python: Fix broken links in notebooks

### PR DESCRIPTION
**Pull request by tomoam**
### Motivation and Context

I have fixed broken links found in the `06-memory-and-embeddings.ipynb` in `notebooks` (both in the `python` and `dotnet`).

### Description

please see the `Files Changed` tab.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/notebooks/06-memory-and-embeddings.ipynb [hunk 1]
```diff
@@ -11,10 +11,10 @@
     "So far, we've mostly been treating the kernel as a stateless orchestration engine.\n",
     "We send text into a model API and receive text out.\n",
     "\n",
-    "In a [previous notebook](04-context-variables-chat.ipynb), we used `context variables` to pass in additional\n",
-    "text into prompts to enrich them with more context. This allowed us to create a basic chat experience.\n",
+    "In a [previous notebook](04-kernel-arguments-chat.ipynb), we used `kernel arguments` to pass in additional\n",
+    "text into prompts to enrich them with more data. This allowed us to create a basic chat experience.\n",
```
Comment (tomoam) [inline python/notebooks/06-memory-and-embeddings.ipynb:15]: This change in this line is not a fix link, but a change in wording. It was changed to match the [06-memory-and-embedding in `dotnet`](https://github.com/microsoft/semantic-kernel/blob/14a0881989e2bd23c75568fe196c705f9bb65f79/dotnet/notebooks/06-memory-and-embeddings.ipynb#L14).

File: python/notebooks/06-memory-and-embeddings.ipynb [hunk 2]
```diff
@@ -11,10 +11,10 @@
...
     "\n",
-    "However, if you solely relied on context variables, you would quickly realize that eventually your prompt\n",
+    "However, if you solely relied on kernel arguments, you would quickly realize that eventually your prompt\n",
```
Comment (tomoam) [inline python/notebooks/06-memory-and-embeddings.ipynb:17]: This change in this line is not a fix link, but a change in wording. It was changed to match the [06-memory-and-embedding in dotnet](https://github.com/microsoft/semantic-kernel/blob/14a0881989e2bd23c75568fe196c705f9bb65f79/dotnet/notebooks/06-memory-and-embeddings.ipynb#L16).

**Comment by tomoam** [comment]
@microsoft-github-policy-service agree

**Review by moonbox3 — APPROVED** [review]

File: python/notebooks/06-memory-and-embeddings.ipynb (hunk 1 above)
Comment (MediVik333) [reply python/notebooks/06-memory-and-embeddings.ipynb:15]: Ok

File: python/notebooks/06-memory-and-embeddings.ipynb (hunk 2 above)
Comment (MediVik333) [reply python/notebooks/06-memory-and-embeddings.ipynb:17]: Sorry, and thanks

**Review by eavanvalkenburg — APPROVED** [review]

**Comment by moonbox3** [comment]
@markwallace-microsoft could you please have a look at the dotnet related code?

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/0ad46052cc9225cded08900d3ba7838c4f5968b2/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5398</b></td><td><b>1007</b></td><td><b>81%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1200 | 11 :zzz: | 0 :x: | 0 :fire: | 24.002s :stopwatch: |

**Review by markwallace-microsoft — APPROVED** [review]

**Comment by tomoam** [comment]
thank you!

**Pull request merged**